cmd/agent/common/loader.go

```diff
@@ -39,17 +38,16 @@ func GetWorkloadmetaInit() workloadmeta.InitHelper {
 
 // LoadComponents configures several common Agent components:
 // tagger, collector, scheduler and autodiscovery
-func LoadComponents(senderManager sender.SenderManager,
-	secretResolver secrets.Component,
-	wmeta workloadmeta.Component,
-	ac autodiscovery.Component,
-	confdPath string) {
+func LoadComponents(secretResolver secrets.Component, wmeta workloadmeta.Component, ac autodiscovery.Component, confdPath string) {
 	confSearchPaths := []string{
 		confdPath,
 		filepath.Join(path.GetDistPath(), "conf.d"),
 		"",
 	}
 
 	// setup autodiscovery. must be done after the tagger is initialized.
+	if !ac.IsStarted() {
```

Comment:
Can you point to the code which assumes the tagger is run before autodiscovery?
If autodiscovery takes the tagger as a parameter, the tagger will be started before autodiscovery.

Reply:
I updated comment. Tagger will always be started before autodiscovery. But autodiscovery component is not started in CLI mode (agent flare and diagnose with force local). It is being loaded in later stage: [pkg/diagnose/check.go](https://github.com/DataDog/datadog-agent/pull/23116/files#diff-c792372d6a3476ee5cf3099091bfd5ce07d3f7223bd4d4627a75f88444a380b8R112)